modules/services/utils/src/main/java/au/com/shiftyjelly/pocketcasts/utils/SharingUrlTimestampParser.kt

```diff
@@ -0,0 +1,37 @@
+package au.com.shiftyjelly.pocketcasts.utils
+
+import kotlin.time.Duration
+import kotlin.time.Duration.Companion.seconds
+
+class SharingUrlTimestampParser {
+    fun parseTimestamp(t: String): Pair<Duration?, Duration?> {
+        val intervalPattern = Regex("""(\d*\.?\d*)?,?(\d*\.?\d*)?""") // t=10,20 or t=,20 or t=10
+        val hmsPattern = Regex("""(?:(\d+)h)?(?:(\d+)m)?(?:(\d+(?:\.\d+)?)s)?""") // t=10h20m30s or t=10m30s or or 10m or t=10s
+        val hhmmssFractionPattern = Regex("""(\d+):(\d{2}):(\d{2})(?:\.(\d+))?""") // t=10:20:30.456
```

Comment:
This should handle mixed patterns. For example `120,0:02:01.5` is valid. Adding test for these would be great.

Reply:
I am struggling to implement the mixed patterns with my limited regex knowledge. I will give another try tomorrow

Reply:
Right, that's why I'm suggesting to have patterns only for single parts. More or less something like this.

```kotlin
fun parseTimestamp(t: String): Pair<Duration, Duration>? {
    val parts = t.split(",")
    if (parts.isEmpty() || parts.size > 2) {
        return null
    }
    val start = parts[0]
    val end = part.getOrNull(1)
    TODO("Parse start and end separately")
}
```

Reply:
Yeah. I was trying to do this, but I faced with some issues. Anyways, please, check here: [5bc9733](https://github.com/Automattic/pocket-casts-android/pull/2235/commits/5bc9733fb4358d2e589df3d9d3643e3510209ba3)